### PR TITLE
Travis cd

### PR DESCRIPTION
**Pull request by nonword**
This PR:

 - Adds a deploy script for QA
 - Adds versioned config for the three deployments. (All existing config is either non-sensitive or encrypted, with one noted exception.)
 - Adds a travis-deploy script so that the existing `deploy-[env]` command can be used by Travis for CD
 - Fixes one test that failed due to missing valid config. (The call is now mocked so that tests do not require valid config.)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,26 @@
 language: node_js
+node_js:
+- '6.10'
+install: npm install
+script: npm test
 cache:
   directories:
-    - "node_modules"
-after_success: npm run coverage
```
Comment (danamansana) [inline .travis.yml]: Why are we deleting coverage?

File: config/development.env [hunk 2]
```diff
@@ -0,0 +1,13 @@
+NODE_ENV=production
+OAUTH_CLIENT_ID=AQECAHjqALewp8JBJNxIQvR4oY795dyG7INaGR1glMsTEgetggAAAHMwcQYJKoZIhvcNAQcGoGQwYgIBADBdBgkqhkiG9w0BBwEwHgYJYIZIAWUDBAEuMBEEDGM1ccp/vwImDcIFwQIBEIAwl1AV5yfZ/tquR4OY9w48ImVdZNi3wXluoY6LqyMDB5s1scHksWy1dmtZLjXAhRi8
+OAUTH_PROVIDER_URL=https://isso.nypl.org/oauth/token
+NYPL_DATA_API_URL=https://platform.nypl.org/api/v0.1/
```
Comment (danamansana) [inline config/development.env]: Are we sure NYPL_DATA_API_URL is just for getting info?

File: config/development.env [hunk 3]
```diff
@@ -0,0 +1,13 @@
+NODE_ENV=production
```
Comment (danamansana) [inline config/development.env]: Is NODE_ENV=production right?

File: package.json [hunk 4]
```diff
@@ -5,16 +5,10 @@
...
-    "setup-dev-env": "mkdir -p config && cp ./sample/deployment.env.sample ./config/dev.env",
-    "setup-prod-env": "mkdir -p config && cp ./sample/deployment.env.sample ./config/prod.env",
-    "setup-dev-sources": "cp ./sample/sample_event_sources.json ./config/event_sources_dev.json",
-    "setup-prod-sources": "cp ./sample/sample_event_sources.json ./config/event_sources_prod.json",
+    "run-local": "./node_modules/.bin/node-lambda run -f ./config/local.env -j event.json",
+    "deploy-development": "./node_modules/.bin/node-lambda deploy -e development -f ./config/development.env -S config/event_sources_development.json --role arn:aws:iam::224280085904:role/lambda_basic_execution --profile nypl-sandbox",
+    "deploy-qa": "./node_modules/.bin/node-lambda deploy -e qa -f ./config/qa.env -S config/event_sources_qa.json --role arn:aws:iam::946183545209:role/lambda-full-access --profile nypl-digital-dev",
+    "deploy-production": "./node_modules/.bin/node-lambda deploy -e production -f ./config/production.env -S config/event_sources_production.json --role arn:aws:iam::946183545209:role/lambda-full-access --profile nypl-digital-dev",
```
Comment (danamansana) [inline package.json]: No VPC?

File: config/development.env (hunk 2 above)
Comment (gkallenberg) [reply config/development.env]: I believe this service does look ups for patron and item record information to pass along to ReCAP's API. For development environments, the URL should be https://dev-platform.nypl.org/api/v0.1.

File: config/qa.env [hunk 5]
```diff
@@ -0,0 +1,15 @@
+NODE_ENV=production
+OAUTH_CLIENT_ID=AQECAHh7ea2tyZ6phZgT4B9BDKwguhlFtRC6hgt+7HbmeFsrsgAAAHMwcQYJKoZIhvcNAQcGoGQwYgIBADBdBgkqhkiG9w0BBwEwHgYJYIZIAWUDBAEuMBEEDGjgjo/vQ4NAC/lq9gIBEIAwI7RrzwumIX3u0I0QD4LO9z8uhtgIuU75+FH5XeQl0xE42HIm2n8Mn1rkXry6TR0N
+OAUTH_PROVIDER_URL=https://isso.nypl.org/oauth/token
+NYPL_DATA_API_URL=https://platform.nypl.org/api/v0.1/
```
Comment (gkallenberg) [inline config/qa.env]: This should also point to https://dev-platform.nypl.org/api/v0.1/

Comment (nonword) [reply config/qa.env]: you mean `qa-platform.nypl..` right?

File: package.json (hunk 4 above)
Comment (nonword) [reply package.json]: No. I believe there's no reason to put a service that doesn't connect to a database inside a VPC.

File: config/development.env (hunk 3 above)
Comment (nonword) [reply config/development.env]: Yes. All deployed code in any environment should operate as NODE_ENV=production. The only place that sometimes changes is when running tests (sometimes convenient to set NODE_ENV=test) or when running locally. NODE_ENV has little to do with deployment environment, which confusingly uses similar terms.

File: config/development.env [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+OAUTH_CLIENT_ID=AQECAHjqALewp8JBJNxIQvR4oY795dyG7INaGR1glMsTEgetggAAAHMwcQYJKoZIhvcNAQcGoGQwYgIBADBdBgkqhkiG9w0BBwEwHgYJYIZIAWUDBAEuMBEEDGM1ccp/vwImDcIFwQIBEIAwl1AV5yfZ/tquR4OY9w48ImVdZNi3wXluoY6LqyMDB5s1scHksWy1dmtZLjXAhRi8
+OAUTH_PROVIDER_URL=https://isso.nypl.org/oauth/token
+NYPL_DATA_API_URL=https://qa-platform.nypl.org/api/v0.1/
```
Comment (danamansana) [inline config/development.env]: shouldn't this be dev-platform?

File: config/qa.env (hunk 5 above)
Comment (gkallenberg) [reply config/qa.env]: Yes -- am still living in "we don't have qa set up land."

**Comment by nonword** [comment]
I believe all requested changes have been made [?]. Thanks for taking a look.

**Pull request merged**